command/ibft/ibft_switch.go

```diff
@@ -0,0 +1,247 @@
+package ibft
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/0xPolygon/polygon-sdk/chain"
+	"github.com/0xPolygon/polygon-sdk/command/helper"
+	"github.com/0xPolygon/polygon-sdk/consensus/ibft"
+	"github.com/0xPolygon/polygon-sdk/helper/common"
+	"github.com/0xPolygon/polygon-sdk/types"
+)
+
+// IBFTSwitchCommand is the command to switch consensus
+type IBFTSwitchCommand struct {
+	helper.Base
+	Formatter *helper.FormatterFlag
+}
+
+func (c *IBFTSwitchCommand) DefineFlags() {
+	c.Base.DefineFlags(c.Formatter)
+
+	c.FlagMap["chain"] = helper.FlagDescriptor{
+		Description: fmt.Sprintf("Specifies the genesis file to update. Default: %s", helper.DefaultConfig().Chain),
+		Arguments: []string{
+			"GENESIS_FILE",
+		},
+		FlagOptional: true,
+	}
+
+	c.FlagMap["type"] = helper.FlagDescriptor{
+		Description: "Sets the new IBFT type [PoA, PoS]",
+		Arguments: []string{
+			"TYPE",
+		},
+		FlagOptional:      false,
+		ArgumentsOptional: false,
+	}
+
+	c.FlagMap["deployment"] = helper.FlagDescriptor{
+		Description: "Sets the height to deploy the contract in PoS",
+		Arguments: []string{
+			"DEPLOYMENT",
+		},
+		FlagOptional:      true,
+		ArgumentsOptional: false,
+	}
+
+	c.FlagMap["from"] = helper.FlagDescriptor{
+		Description: "Sets the height to switch the new type",
+		Arguments: []string{
+			"FROM",
+		},
+		FlagOptional:      false,
+		ArgumentsOptional: false,
+	}
+}
+
+// GetHelperText returns a simple description of the command
+func (c *IBFTSwitchCommand) GetHelperText() string {
+	return "Add settings in genesis.json to switch IBFT type"
+}
+
+func (c *IBFTSwitchCommand) GetBaseCommand() string {
+	return "ibft switch"
+}
+
+// Help implements the cli.PeersAdd interface
+func (c *IBFTSwitchCommand) Help() string {
+	c.DefineFlags()
+
+	return helper.GenerateHelp(c.Synopsis(), helper.GenerateUsage(c.GetBaseCommand(), c.FlagMap), c.FlagMap)
+}
+
+// Synopsis implements the cli.PeersAdd interface
+func (c *IBFTSwitchCommand) Synopsis() string {
+	return c.GetHelperText()
+}
+
+// Run implements the cli.PeersAdd interface
+func (c *IBFTSwitchCommand) Run(args []string) int {
+	flags := c.Base.NewFlagSet(c.GetBaseCommand(), c.Formatter)
+
+	var genesisPath, rawType, rawDeployment, rawFrom string
+
+	flags.StringVar(&genesisPath, "chain", helper.DefaultConfig().Chain, "")
+	flags.StringVar(&rawType, "type", "", "")
+	flags.StringVar(&rawDeployment, "deployment", "", "")
+	flags.StringVar(&rawFrom, "from", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		c.Formatter.OutputError(err)
+
+		return 1
+	}
+
+	typ, err := ibft.ParseType(rawType)
```

Comment:
Typo: `typ`

Reply:
tbh, it's not typo. I didn't want to use reserved word. Renamed at